binderhub/app.py

```diff
@@ -96,6 +97,23 @@ def _valid_base_url(self, proposal):
             proposal.value = proposal.value + '/'
         return proposal.value
 
+    auth_enabled = Bool(
+        False,
+        help="""If JupyterHub authentication enabled, 
+        require user to login (don't create temporary users during launch) and 
+        start the new server for the logged in user.""",
+        config=True)
+
+    use_oauth = Bool(
```

Comment:
It seems we don't nee to have separate `auth_enabled` and `use_oauth` flags. We can have a single `auth_enabled`  flag that sets up `HubOAuth`, perhaps? Or even run with a `jupyterhub_service` flag that will load the full service config (URL, etc.)?

Reply:
but don't we need 2 flags to distinguish if an OAuthenticator or another authenticator (e.g. `ldapauthenticator`, `jhub_shibboleth_auth`) is used?

Reply:
I mean that when I use `jhub_shibboleth_auth`, `auth_enabled` is True but `user_oauth` is False. And then `HubAuth` is used (not `HubOAuth`).